programs/ui-wrapper/src/processors/place_order.rs

```diff
@@ -169,18 +169,20 @@ pub(crate) fn process_place_order(
         )?;
     }
 
-    // Call expand so claim seat has enough free space
-    // and owner doesn't get charged rent
-    // TODO: could check if needed before
-    invoke(
-        &expand_market_instruction(market.key, payer.key),
-        &[
-            manifest_program.info.clone(),
-            payer.info.clone(),
-            market.info.clone(),
-            system_program.info.clone(),
-        ],
-    )?;
+    // Call expand so claim seat has enough free space and owner doesn't get
+    // charged rent. This is done here to keep payer and owner separate in the
+    // case of PDA owners.
+    if !market.get_fixed()?.has_free_block() {
```

Comment:
this didn't work for me in certain scenarios, if the market has 1 free block and we place a new order, it will still need to expand after place for the swap slot. we would need has_two_free_blocks